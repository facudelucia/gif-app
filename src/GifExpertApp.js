import React, { useState } from 'react';
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h2>GifRandomApp</h2>
            <hr />
            <AddCategory setCategories={setCategories} />
            {categories.map((category) => (
                <GifGrid key={category} category={category} />
            ))}
        </>
    );
}

export default GifExpertApp;