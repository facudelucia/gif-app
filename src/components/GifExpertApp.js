import React,{Fragment, useState} from 'react';
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = () => {
    const [categories, setCategories] = useState(["Welcome"])
  
    return ( 
        <Fragment>
            <h2>GifRandomApp</h2>
            <hr />
            <AddCategory setCategories={setCategories}/>
            {categories.map((category) => {
                return <GifGrid key={category} category={category}/>
            })}
        </Fragment>
     );
}
 
export default GifExpertApp;