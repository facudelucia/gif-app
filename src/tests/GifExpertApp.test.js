import React from 'react'
import { shallow } from "enzyme"
import GifExpertApp from '../GifExpertApp'
import '@testing-library/jest-dom'



describe('Pruebas en <GifExpertApp />', () => {

    test('Debe mostrar el componente correctamente', () => {
        let wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })
    test('Debe mostrarse una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })

})