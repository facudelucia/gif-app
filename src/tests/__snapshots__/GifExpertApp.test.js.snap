// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifRandomApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrarse una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifRandomApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;
